### PR TITLE
AWS Cloudformation proposal

### PR DESCRIPTION
**Pull request by stvhawes**

### CODE REVIEW

**Comment by stvhawes** [comment]
Hi, I hope this meets contributing standards; feel free to edit.

File: aws/paypal-ipn-cloudformation.yml [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    Default: timeout-ipn-test
+  ErrorQueueName:
+    Description: Delivery Error Queue
+    Type: String
+    Default: errors-ipn-test
+  VerifyURL:
+    Description: PaypalMessageVerificationURL
+    Type: String
+    Default: https://ipnpb.sandbox.paypal.com/cgi-bin/webscr   # for Sandbox IPNs
+    #    Default: https://ipnpb.paypal.com/cgi-bin/webscr   # for live IPNs
```
Comment (overint) [inline aws/paypal-ipn-cloudformation.yml]: Maybe add a comment to the readme about switching this?

File: aws/paypal-ipn-cloudformation.yml [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            - '  containedMessage = event["Records"][0]["Sns"]["Message"]'
+            - '  bodyText = json.loads(containedMessage)["body-text"]'
+            - !Sub '  checkQuery = "${VerifyURL}?cmd=_notify-validate&"+bodyText'
+            - '  verifyResponse = urlopen(checkQuery).read().strip()'
+            - '  print("verifyResponse: " + verifyResponse)'
+            - '  if verifyResponse == "VERIFIED":'
+            - '    print("VERIFIED")'
+            - !Join 
+              - ''
+              - - '    response = sqs.send_message(QueueUrl="https://sqs.ap-southeast-2.amazonaws.com/127569160727/'
```
Comment (overint) [inline aws/paypal-ipn-cloudformation.yml]: Could we maybe switch these urls to `<change me>` or similar?

File: aws/paypal-ipn-cloudformation.yml [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+                  "resource-path" : "$context.resourcePath"
+              }              
+            }
+      MethodResponses:
+      - StatusCode: 200
+        ResponseModels:
+          text/html: Empty
+
+# Stages
+  # Invoke URL: https://i681ypmtg2.execute-api.ap-southeast-2.amazonaws.com/dev
```
Comment (overint) [inline aws/paypal-ipn-cloudformation.yml]: As above.

**Comment by stvhawes** [comment]
thanks overint!  I'll fix those items + retest

File: aws/paypal-ipn-cloudformation.yml (hunk 1 above)
Comment (stvhawes) [reply aws/paypal-ipn-cloudformation.yml]: done.  added a section describing each parameter.

File: aws/paypal-ipn-cloudformation.yml (hunk 2 above)
Comment (stvhawes) [reply aws/paypal-ipn-cloudformation.yml]: parameterised them, thanks for the pickup.

File: aws/paypal-ipn-cloudformation.yml (hunk 3 above)
Comment (stvhawes) [reply aws/paypal-ipn-cloudformation.yml]: as above.

**Comment by stvhawes** [comment]
some other changes added + happy path tested.
* tightened security policies sending to SNS and SQS
* added the throttles from the README and updated the readme with more throttle detail

**Comment by overint** [comment]
Thanks for making those changes!

**Pull request merged**